### DISCUSSION TITLE
improve container interconnectability

### DISCUSSION THREAD

**Discussion by the-leonat** [coollabsio/coolify#1910] [Feature Requests]
At the moment one container can connect to the other (if on the same network) by assigning a static ip when attaching the container to the docker network with the `--ip` custom docker option. 
Rolling deployments are not longer possible and thus if the deployment failes the service is not available, which is not good at all. 
But it makes sense as containers during deployment need to coexist (the old running one, the new one to deploy) in order for rolling deployments to work.
I suggest to implement the `docker run` parameter `--network-alias` which would resolve this issue, as multiple containers can have the same alias on the same network. We could then reference containers by their network alias instead of their internal ip respectively.

**Comment by indieTrav** [comment]
Yes! this is needed.

**Comment by mt-bt** [comment]
+1

**Comment by the-leonat** [comment]
just wanted to check in, if this will be implemented anytime soon.  @andrasbacsai

**Comment by stevenirby** [comment]
Maybe an environmental variable could be used for this? I would take that over how it works now.

**Comment by piotr-woojcik** [comment]
I need that feature also so I made pull request: https://github.com/coollabsio/coolify/pull/4741

**Comment by SegoviaDiego** [comment]
This is a must-have feature! 🙏 

@andrasbacsai are there any updates regarding the open PR? #4741 

We are trying to deploy coolify to production with a cluster of 6 VPSs, but currently it is not possible without losing rolling updates.

**Comment by andrasbacsai** [comment]
The PR/feature will be released in the next version (v407).

**Reply by SegoviaDiego** [reply]
Thank you so much!! ❤️
